### ISSUE TITLE
Question about with skiping handlers while rendering partials

### ISSUE THREAD

**Issue by kortirso** [thoughtbot/props_template#27]
There is [line](https://github.com/thoughtbot/props_template/blob/main/lib/props_template/extensions/partial_renderer.rb#L85) with `pass_opts.delete(:handlers)` while rendering partial

Situation
- there are jbuilder and props handlers in the app (because it's migration time from jbuilder to props)
- rendering in template `json.array! orders, partial: ["shared/order_table_item", handlers: [:props]]`

Issue, in such case
- PropsTemplate skips `handlers` option
- ActionView uses all available handlers and select jbuilder for rendering

Why is there such `pass_opts.delete(:handlers)` @jho406 ? I can overwrite this method of course, but I don't see any reason to skip handlers

**Comment by jho406** [comment]
Hi @kortirso,

Just some background: Whenever we use `json.array! do ...end ` or `json.foo do ... end`, `props_template` actually builds the JSON incrementally using [push_array](https://www.rubydoc.info/gems/oj/Oj/StringWriter#push_array-instance_method)  or similarly, [push_json](https://www.rubydoc.info/gems/oj/Oj/StringWriter#push_json-instance_method) from OJ. 

This is different from jbuilder, which builds a ruby hash first, finally as a last step, passes the entire ruby hash to OJ to render to JSON.

In other words, `props_template` views can only be used with `props_template` partials because of that fundamental difference. The deletion of the handler makes sure that happens. Migrating from jbuilder to props_template would require you to migrate one full view at a time supported by tests.

Stepping back, could you describe how you're migrating from jbuilder to props_template? Maybe if we start there I can give some concrete recommendations.

**Comment by kortirso** [comment]
For example, right now there are jbuilder and props views for some action, feature flag defines what serializer will be used, so in controller we have

`render handlers: [response_serializer]`, values can be :jbuilder or :props

On that step everything is fine, because during template rendering PropsTemplate allows to set handlers option

But inside template there could be rendering of partial (that exists for jbuilder and props)

```
json.array! orders, partial: ["shared/order", locals: {employee: employee}] do
end
```

And here partial does not accept handlers variable because it will be deleted later, so active view will try to use any of registered handlers and will choose jbuilder

So as temporary solution I overwrite method `block_opts_to_render_opts` with `pass_opts[:handlers] = [:props]` instead of deleting

**Comment by jho406** [comment]
> For example, right now there are jbuilder and props views for some action, feature flag defines what serializer will be used.

Ah! That's a great approach if the partials from props_template and jbuilder are the syntactically same. 

> So as temporary solution I overwrite method block_opts_to_render_opts with pass_opts[:handlers] = [:props] instead of deleting

I would be open to making that a permanent change as its still compatible with the reasoning I outlined above. Instead of deleting and letting actionview use its default behavior, we can make it explicit. Would you be open to making a PR?
